lib/convection/model/diff.rb

```diff
@@ -12,23 +13,32 @@ class Diff
       attr_reader :action
       attr_reader :ours
       attr_reader :theirs
-      colorize :action, :green => [:create], :yellow => [:update], :red => [:delete]
+      colorize :action, :green => [:create], :yellow => [:update], :red => [:delete, :replace]
 
       def initialize(key, ours, theirs)
         @key = key
         @ours = ours
         @theirs = theirs
 
-        @action = if ours && theirs then :update
-                  elsif ours then :create
-                  else :delete
+        @action = if ours && theirs
+                    property_name = key[/AWS::[A-Za-z0-9:]+\.[A-Za-z0-9]+/]
```

Comment:
Really nice use of a regex index here. I love when you find a good use case for this.